### DISCUSSION TITLE
Why does onEdgeUpdateStart sends the opposite handleType?

### DISCUSSION THREAD

**Discussion by bodinsamuel** [xyflow/xyflow#3201] [Q&A]
Hey,

When using `onEdgeUpdateStart` we receive the edge and handleType but I'm not sure why the `handleType` is the opposite of the handle we picked

When picking the source it sends `target`, and when picking the target it sends `source`; its the opposite of what we are modifying.
Cf: https://capture.dropbox.com/xV6aHmC9WCHnomc1

https://github.com/wbkd/react-flow/assets/1637651/9c884373-4c45-4d3f-9375-2176f61f8c27

Is this normal?
Thanks

**Comment by bcakmakoglu** [comment]
The handle type you see is the one that is still connected to the edge when starting an edge update - not the handle you triggered the edge update from.

An edge update basically just replaces the edge temporarily with a connection line, which needs to know which handle "started" the connection and what type it is and that's the same info you get fed through the `onEdgeUpdateStart` cb.
